src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -21,22 +34,34 @@ internal static unsafe IPAddress Parse(string ipString, bool tryParse)
                 }
                 throw new ArgumentNullException(nameof(ipString));
             }
+ 
+            return Parse(ipString.ToCharArray(), tryParse);
+        }
 
-            if (ipString.IndexOf(':') != -1)
+        internal static unsafe IPAddress Parse(ReadOnlySpan<char> ipSpan, bool tryParse)
+        {
+            return Parse(ipSpan, tryParse);
```

Comment:
This seems to have worked by accident because of the parameters. Renaming it to avoid issues.